sechub-commons-archive/src/main/java/com/mercedesbenz/sechub/commons/archive/ArchiveSupport.java

```diff
@@ -508,7 +491,7 @@ private ArchiveExtractionResult extract(ArchiveInputStream sourceArchiveInputStr
                     continue;
                 }
                 try (OutputStream outputFileStream = new FileOutputStream(outputFile)) {
-                    IOUtils.copy(sourceArchiveInputStream, outputFileStream);
+                    IOUtils.copy(safeArchiveInputStream, outputFileStream);
```

Comment:
## I thought about following:
The tests do check the implementations of `FileZip`, `SafeArchiveIpustream` etc. All well done tested, but what we do not test is, that extracting with `ArchiveSupport` does always use a `SafeARchiveInputStream` at the end/ the archive support is using the implemented protection against Zip bombs at all.

## We could do this following ways IMO:

a) we write an integration test with a zimb bomb upload via rest directly - and check for upload error without upload - one problem, e.g. too many directories - would  be enough. But this could (maybe) lead to a longer running integrtion test? (and we do not want to increase build time much)

b) we could wrap here the static call via IOUtils - e.g. by a package private field using a `IOUTilsCopyWrapper` (or a better interface name...) with a default implementation which uses IOUtil.copy... Inside the test we could mock it with `Mockito` and check if it is called with a a `SafeArchiveInputStream`.

I think we should implement at least one of the possibilities.